circuits/circom/trees/incrementalQuinTree.circom

```diff
@@ -0,0 +1,249 @@
+include "../../node_modules/circomlib/circuits/mux1.circom";
+include "../../node_modules/circomlib/circuits/comparators.circom";
+include "../hasherPoseidon.circom";
+include "./calculateTotal.circom";
+include "./checkRoot.circom";
+
+// This file contains circuits for quintary Merkle tree verifcation.
+// It assumes that each node contains 5 leaves, as we use the PoseidonT6
+// circuit to hash leaves, which supports up to 5 input elements.
+
+/*
+Note: circom has some particularities which limit the code patterns we can use.
+- You can only assign a value to a signal once.
+- A component's input signal must only be wired to another component's output
+  signal.
+- Variables are only used for loops, declaring sizes of things, and anything
+  that is not related to inputs of a circuit.
+- You can't use a signal as a list index.
+*/
+
+
+/*
+ * Given a list of items and an index, output the item at the position denoted
+ * by the index. The number of items must be less than 8, and the index must
+ * be less than the number of items.
+ */
+template QuinSelector(choices) {
+    signal input in[choices];
+    signal input index;
+    signal output out;
+    
+    // Ensure that index < choices
+    component lessThan = LessThan(3);
+    lessThan.in[0] <== index;
+    lessThan.in[1] <== choices;
+    lessThan.out === 1;
```

Comment:
These are values provided by the coordinator, right? If yes, this condition is fine :)

Reply:
Yup!